### PR TITLE
Updating README, bug fixes & additional endpoints

### PR DESCRIPTION
**Pull request by iancrowther**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,14 +14,15 @@ module.exports = function (options) {
 
   return function send ({endpoint, tokens, config, cb}) {
     const _config = configFactory(config)
+    const _tokens = tokens || []
```
Comment (confuser) [inline index.js]: Would it not be better to leverage ES6 default parameters?

File: lib/end-points.js [hunk 2]
```diff
@@ -1,130 +1,222 @@
 /* jslint maxlen: 200 */
```
Comment (confuser) [inline lib/end-points.js]: This file is auto generated via https://github.com/confuser/node-dotmailer/blob/master/lib/end-point-builder.js

File: index.js (hunk 1 above)
Comment (iancrowther) [reply index.js]: ok no probs

File: lib/end-points.js (hunk 2 above)
Comment (iancrowther) [reply lib/end-points.js]: oh right!

Comment (iancrowther) [reply lib/end-points.js]: will review, verify the endpoints and update

File: index.js (hunk 1 above)
Comment (iancrowther) [reply index.js]: updated

File: lib/end-points.js (hunk 2 above)
Comment (iancrowther) [reply lib/end-points.js]: re-generated - nice touch btw!

File: README.md [hunk 3]
```diff
@@ -12,15 +12,27 @@ npm install dotmailer --save
 ```
 
 ## Usage
-```dotMailer(ApiFunction, tokens..., dataObject, callback)```
+```dotMailer({endpoint, tokens, config, cb})```
```
Comment (confuser) [inline README.md]:
```suggestion
```dotMailer({endpoint, tokens, config, callback})```
```
Sorry, being picky but cb may not be overly clear to some, should probably switch to promises at some point anyway

Comment (iancrowther) [reply README.md]: yep happy to update..

File: index.js (hunk 1 above)
Comment (confuser) [reply index.js]: Looks like the builder needs updating to put the `,` in the correct place, sorry about that 😞

File: README.md (hunk 3 above)
Comment (iancrowther) [reply README.md]: ha.. Promises & TS were coming in a PR in the next couple of wks

File: lib/end-points.js (hunk 2 above)
Comment (confuser) [reply lib/end-points.js]: Looks like the builder needs updating to put the `,` in the correct place, sorry about that 😞

File: README.md (hunk 3 above)
Comment (iancrowther) [reply README.md]: done

Comment (confuser) [reply README.md]: Promises 👍  Not fond of TypeScript though, would prefer to keep it JS; happy to accept a TS typings doc if possible though

**Comment by iancrowther** [comment]
image

File: lib/end-points.js (hunk 2 above)
Comment (iancrowther) [reply lib/end-points.js]: yeah no worries - i get the comment about manual steps now ;-)

**Comment by confuser** [comment]
Published as `v3.0.1`. Thanks!

**Pull request merged**